Assets/Scripts/Enemy/EnemyController.cs

```diff
@@ -0,0 +1,234 @@
+        private float wanderTimer;
+
+        [Tooltip("Whether the area should be fixed on enable.")] [SerializeField]
+        public bool freezeArea;
+
+        [Tooltip("The ground the NPC should walk on.")] [SerializeField]
+        private LayerMask moveLayer;
+
+        [Header("Attacking")]
+        [Tooltip("Determine whether the NPC attacks and chases the player.")] [SerializeField]
+        private bool isAggressive;
+        
+        [Tooltip("The speed the NPC chases targets with.")] [SerializeField]
+        private float runningSpeed;
+        
+        [Tooltip("The speed the NPC rotates with.")] [SerializeField]
+        private float rotationSpeed;
+        
+        [Tooltip(
+            "The maximum difference in degrees for the NPC between look direction and target direction in order to be facing the target.")]
+        [SerializeField]
+        private float rotationTolerance;
+        
+        [Tooltip("The radius around the NPC in which a player can escape.")] [SerializeField]
+        private float escapeRadius;
+
+        [Tooltip("Show escape radius in scene view.")] [SerializeField]
+        private bool showEscapeRadius;
+
+        [Tooltip("TargetFinderComponent - Finds nearby targets")] [SerializeField]
+        private TargetFinder targetFinder;
+        
+        
+        
+        // component references
+        private NavMeshAgent _agent;
+
+        private WanderAI _wanderAI;
+        private AttackLogic _attackLogic;
+
+        // hold player object to check distance
+        private float _timer;
+        private bool _isChasing;
+        private float _initialSpeed;
+
+        private void Awake()
+        {
+            // init components
+            _agent = GetComponent<NavMeshAgent>();
+            _wanderAI = new WanderAI();
+            _attackLogic = GetComponent<AttackLogic>();
+            if (_attackLogic == null)
+                Debug.LogError("NPC is not able to attack without an AttackLogic component.");
+        }
+
+        /// <summary>
+        /// Gets the player object and freezes wander area if necessary.
+        /// </summary>
+        private void OnEnable()
+        {
+            if (freezeArea)
+                wanderArea.FreezeArea();
+            _timer = wanderTimer;
+            // remember initial speed
+            _initialSpeed = _agent.speed;
+        }
+
+        /// <summary>
+        /// Moves around randomly in the defined wander area at a defined frequency and checks for an attackable 
+        /// target near (which is the player for now). If the player is near, starts chasing (and attacking) it until the aggro is lost.
+        /// </summary>
+        private void Update()
+        {
+            if (!_isChasing)
+                Wander();
+
+            if (!isAggressive || targetFinder.Targets.Count == 0) return;
+
+            // Attack first target in list
+            GameObject currentTarget = targetFinder.Targets[0];
+
+            if (!_isChasing)
```

Comment:
```
if (!isAggressive || targetFinder.Targets.Count == 0) Wander();

// Attack first target in list
GameObject currentTarget = targetFinder.Targets[0];

if (!_isChasing) StartChasing();
```

Reply:
This would prevent an aggressive enemy from wandering, right?

Reply:
I think this has to be reworked later to support passive, neutral and aggressive NPCs